opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1644,6 +1698,13 @@ namespace Opm {
                     cellRes[ flowPhaseToEbosCompIdx( flowPhaseIdx ) ] /= refDens;
                     cellRes[ flowPhaseToEbosCompIdx( flowPhaseIdx ) ] *= cellVolume;
                 }
+                if (has_solvent_) {
+                    // no need to store refDens for all cells?
+                    const auto& intQuants = ebosSimulator_.model().cachedIntensiveQuantities(cellIdx, /*timeIdx=*/0);
```

Comment:
In some sense, yes: It's not a *regression* IMO, so I think it is acceptable even though I'd prefer if this was changed.